.golangci.yml

```diff
@@ -0,0 +1,65 @@
+service:
+  golangci-lint-version: 1.23.6 # Keep this in sync with .circleci/config.yml and Jenkinsfile.pr
+
+run:
+  timeout: 5m
+  modules-download-mode: readonly
+
+linters-settings:
+  gofmt:
+    simplify: true
+  govet:
+    check-shadowing: false # set this to true from time to time to check for possible issues
+    disable-all: true
+    enable:
+      - asmdecl      # report mismatches between assembly files and Go declarations
+      - assign       # check for useless assignments
+      - atomic       # check for common mistakes using the sync/atomic package
+      - bools        # check for common mistakes involving boolean operators
+      - buildtag     # check that +build tags are well-formed and correctly located
+      - cgocall      # detect some violations of the cgo pointer passing rules
+      - composites   # check for unkeyed composite literals
+      - copylocks    # check for locks erroneously passed by value
+      - errorsas     # report passing non-pointer or non-error values to errors.As
+      - httpresponse # check for mistakes using HTTP responses
+      - loopclosure  # check references to loop variables from within nested functions
+      - lostcancel   # check cancel func returned by context.WithCancel is called
+      - nilfunc      # check for useless comparisons between functions and nil
+      - printf       # check consistency of Printf format strings and arguments
+      - shift        # check for shifts that equal or exceed the width of the integer
+      - stdmethods   # check signature of methods of well-known interfaces
+      - structtag    # check that struct field tags conform to reflect.StructTag.Get
+      - tests        # check for common mistaken usages of tests and examples
+      - unmarshal    # report passing non-pointer or non-interface values to unmarshal
+      - unreachable  # check for unreachable code
+      - unsafeptr    # check for invalid conversions of uintptr to unsafe.Pointer
+      - unusedresult # check for unused results of calls to some functions
+
+linters:
+  disable-all: true
+  enable:
+    - deadcode
+    - gofmt
+    - golint
+    - gosimple
+    - govet
+    - ineffassign
+    - structcheck
+    - unconvert
+    - unused
+    - varcheck
+    - scopelint
+
+issues:
+  exclude-rules:
+    - linters:
+      # ignore unused warnings from enterprise code
+      # add more as required.
+      - unused
+      text: "RedisSupplier|LocalCacheSupplier|Enterprise"
+
+    - linters:
+      # ignore golint error for a lot of packages for now
+      - golint
```

Comment:
We can remove this too.

Reply:
I think that's to avoid getting ids related errors which we are not fixing yet.

Reply:
Oh my bad. I thought those exceptions were copied from the server repo, but now I see these are from this repo.